ERCS/erc-7628.md

```diff
@@ -0,0 +1,111 @@
+---
+eip: 7628
+title: ERC-721 Balance Extension
+description: Extend ERC-721 to include balance functionalities similar to ERC-20.
+author: Chen Liaoyuan (@chenly) <cly@kip.pro>
+discussions-to: https://ethereum-magicians.org/t/erc-7628-erc-721-balance-extension/18744
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-02-20
+requires: 20, 721
+---
+
+## Abstract
+
+This extension defines an interface that adds balance functionalities to [ERC-721](./eip-721.md) tokens, enabling functions and events for balance querying, transfer, and approval similar to [ERC-20](./eip-20.md) tokens.
+
+## Motivation
+
+Some NFTs represent divisible ownership stakes or shares of associated rights, such as those pertaining to digital assets, real estate, or intellectual property. The ability to accurately query and exchange these shares is crucial for facilitating efficient market operations. However, the absence of built-in support for token balances within the current [ERC-721](./eip-721.md) standard poses challenges and limitations in managing and trading these rights effectively. By introducing functionalities that mirror those of [ERC-20](./eip-20.md) token balances, we can offer a comprehensive solution for [ERC-721](./eip-721.md) tokens. This enhancement empowers holders to effortlessly inquire about and exchange their respective ownership shares, thereby promoting wider acceptance and circulation of NFTs.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+Implementers of this extension **MUST** have all of the following functions:
+
+```solidity
+pragma solidity ^0.8.0;
+
+interface IERC7628 /* is IERC721 */ {
+    /// @notice Returns the number of decimal places used for token balances.
+    /// @return The number of decimal places.
+    function balanceDecimals() external view returns (uint8);
+
+    /// @notice Returns the total sum of token balances in existence.
+    /// @return The total sum of balances for all tokens.
+    function totalBalances() external view returns (uint256);
+
+    /// @notice Returns the balance of the specified token.
+    /// @param tokenId The identifier of the token.
+    /// @return The balance of the token.
+    function balanceOf(uint256 tokenId) external view returns (uint256);
```

Comment:
Does this collide with the function from ERC-721, and if so, is that a problem?

Reply:
The new function is called "shareOf".